From 035707fb15a90ce80d297ddb75b7e908c9fe051d Mon Sep 17 00:00:00 2001
From: Jinkun Hong <jinkun.hong@rock-chips.com>
Date: Mon, 6 Jun 2022 09:46:02 +0800
Subject: [PATCH 2/6] Increase width and height limits

Signed-off-by: Jinkun Hong <jinkun.hong@rock-chips.com>
---
 src/draw/lv_img_buf.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/draw/lv_img_buf.h b/src/draw/lv_img_buf.h
index 984483e..5f26afd 100644
--- a/src/draw/lv_img_buf.h
+++ b/src/draw/lv_img_buf.h
@@ -108,8 +108,8 @@ typedef uint8_t lv_img_cf_t;
 #if LV_BIG_ENDIAN_SYSTEM
 typedef struct {
 
-    uint32_t h : 11; /*Height of the image map*/
-    uint32_t w : 11; /*Width of the image map*/
+    uint32_t h : 16; /*Height of the image map*/
+    uint32_t w : 16; /*Width of the image map*/
     uint32_t reserved : 2; /*Reserved to be used later*/
     uint32_t always_zero : 3; /*It the upper bits of the first byte. Always zero to look like a
                                  non-printable character*/
@@ -125,8 +125,8 @@ typedef struct {
 
     uint32_t reserved : 2; /*Reserved to be used later*/
 
-    uint32_t w : 11; /*Width of the image map*/
-    uint32_t h : 11; /*Height of the image map*/
+    uint32_t w : 16; /*Width of the image map*/
+    uint32_t h : 16; /*Height of the image map*/
 } lv_img_header_t;
 #endif
 
-- 
2.7.4

