From d1ad43ef376e89794b8596c53e6dfe424ca25eda Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 24 Dec 2018 15:38:58 +0800
Subject: [PATCH 04/17] gsttools: videowidget: Use transparent background

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 src/gsttools/qgstreamervideowidget.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gsttools/qgstreamervideowidget.cpp b/src/gsttools/qgstreamervideowidget.cpp
index 91a487e1..6e7def99 100644
--- a/src/gsttools/qgstreamervideowidget.cpp
+++ b/src/gsttools/qgstreamervideowidget.cpp
@@ -56,7 +56,7 @@ public:
     {
         setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
         QPalette palette;
-        palette.setColor(QPalette::Window, Qt::black);
+        palette.setColor(QPalette::Window, Qt::transparent);
         setPalette(palette);
     }
 
-- 
2.20.1

